testsuite/diem-swarm/src/swarm.rs

```diff
@@ -121,6 +121,10 @@ impl DiemNode {
         self.peer_id
     }
 
+    pub fn stop(&mut self) {
+        self.process.kill().expect("node is not running");
+    }
```

Comment:
So this probably won't be sufficient since then we won't ever be able to start back up the Node. We probably want to change the local Node to something like this:
```
#[derive(Debug)]
struct Process(Child);

impl Drop for Process {
    // When the Process struct goes out of scope we need to kill the child process
    fn drop(&mut self) {
        // check if the process has already been terminated
        match self.process.try_wait() {
            // The child process has already terminated, perhaps due to a crash
            Ok(Some(_)) => {}

            // The process is still running so we need to attempt to kill it
            _ => {
                self.process.kill().expect("Process wasn't running");
                self.process.wait().unwrap();
            }
        }
    }
}

struct LocalNode {
    diem_node_bin: PathBuf,
    process: Option<Process>,
    node_id: NodeId,
    peer_id: PeerId,
    log_path: PathBuf,
    config: NodeConfig,
}
```

Notice wrapping the process in an Option as a way to signal if the Node is running or not. It will let us start and stop the node while being able to reuse the Node struct.

Reply:
this is a brilliant way to determine node status